graph/pull_v2.go

```diff
@@ -73,30 +73,29 @@ func (p *v2Puller) pullV2Repository(tag string) (err error) {
 
 	}
 
-	c, err := p.poolAdd("pull", taggedName)
-	if err != nil {
-		if c != nil {
-			// Another pull of the same repository is already taking place; just wait for it to finish
-			p.sf.FormatStatus("", "Repository %s already being pulled by another client. Waiting.", p.repoInfo.CanonicalName)
-			<-c
-			return nil
-		}
-		return err
+	broadcaster, found := p.poolAdd("pull", taggedName)
+	if found {
+		// Another pull of the same repository is already taking place; just wait for it to finish
+		p.config.OutStream.Write(p.sf.FormatStatus("", "Repository %s already being pulled by another client. Waiting.", p.repoInfo.LocalName))
```

Comment:
V1 does not have an equivalent message, but I am questioning whether this is really necessary given the following output is from the pull.